core-api/src/main/java/com/optimizely/ab/config/audience/UserAttribute.java

```diff
@@ -45,25 +49,31 @@ public String getType() {
         return type;
     }
 
+    public String getMatch() {
+        return match;
+    }
+
     public Object getValue() {
         return value;
     }
 
-    public boolean evaluate(Map<String, ?> attributes) {
+    public @Nullable Boolean evaluate(Map<String, ?> attributes) {
         // Valid for primative types, but needs to change when a value is an object or an array
         Object userAttributeValue = attributes.get(name);
 
-        if (value != null) { // if there is a value in the condition
-            // check user attribute value is equal
-            return value.equals(userAttributeValue);
+        if (!"custom_attribute".equals(type)) {
```

Comment:
nit: Why not invoke the `equals` method from the variable itself?

Reply:
Good thinking @thomaszurkan-optimizely!